lib/services/devices/deviceService.js

```diff
@@ -199,6 +357,104 @@ function createInitialEntity(deviceData, newDevice, callback) {
     request(options, createInitialEntityHandler(deviceData, newDevice, callback));
 }
 
+/**
+ * Creates the initial entity representing the device in the Context Broker. This is important mainly to allow the
+ * rest of the updateContext operations to be performed using an UPDATE action instead of an APPEND one.
+ *
+ * @param {Object} deviceData       Object containing all the deviceData needed to send the registration.
+ * @param {Object} newDevice        Device object that will be stored in the database.
+ */
+function createInitialEntity(deviceData, newDevice, callback) {
+    if (config.checkNgsi2()) {
+        createInitialEntityNgsi2(deviceData, newDevice, callback);
+    } else {
+        createInitialEntityNgsi1(deviceData, newDevice, callback);
+    }
+}
+
+/**
+ * Updates the entity representing the device in the Context Broker using NGSIv2.
+ *
+ * @param {Object} deviceData       Object containing all the deviceData needed to send the registration.
+ * @param {Object} updatedDevice    Device object that will be stored in the database.
+ */
+function updateEntityNgsi2(deviceData, updatedDevice, callback) {
+    var options = {
+        url: config.getConfig().contextBroker.url + '/v2/entities/' + String(deviceData.name) + '/attrs',
+        method: 'PUT',
```

Comment:
PUT will replace *all* entiy attributes with the ones you sent from IOTA. I think this is not the desired behaviour... as the entity could be shared (i.e. has attributes out of the scope of IOTA used by other applications) that would be destroyed.

I understand that PATCH should be used.

Reply:
... or maybe POST (the difference is that PATCH raises an error if the attribute doesn't peviously exist, while POST creates it if it doesn't previously exist).

Reply:
Thanks for the comment! I missed that part of the API documentation! :S

Reply:
Fixed in a34df7f50e3d7a185ddb26401965f6f031c69fa5